eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala

```diff
@@ -80,7 +80,16 @@ class LocalChannelKeyManager(seed: ByteVector, chainHash: ByteVector32) extends
     DeterministicWallet.KeyPath(Seq(next(), next(), next(), next(), next(), next(), next(), next(), last))
   }
 
-  override def fundingPublicKey(channelKeyPath: DeterministicWallet.KeyPath): ExtendedPublicKey = publicKeys.get(internalKeyPath(channelKeyPath, hardened(0)))
+  override def fundingPublicKey(channelKeyPath: DeterministicWallet.KeyPath, fundingTxIndex: Long): ExtendedPublicKey = {
+    val baseFundingKeyPath = internalKeyPath(channelKeyPath, hardened(0))
+    val keyPath = if (fundingTxIndex == 0) {
+      // For backward-compat with pre-splice channels, the initial funding tx of a channel use the base keypath with no derivation
+      baseFundingKeyPath
+    } else {
+      baseFundingKeyPath.derive(fundingTxIndex)
+    }
```

Comment:
⚠️  This is how we achieve backward compat. Sensitive code!

Reply:
We can easily work around this issue I think, by using a different path for dynamic funding pubkeys (e.g. `internalKeyPath(channelKeyPath, hardened(6))`). It feels also less hacky.

Reply:
Yes, that would be a good idea!

Reply:
Done in [34ba18b](https://github.com/ACINQ/eclair/pull/2634/commits/34ba18bae5461e547be1b7a8ee4705d992364a91).

Reply:
Perfect!